### PR TITLE
Tab completion (#204)

### LINKED ISSUE #204: Tab completion for arguments
## Background
Many command-line tools have tab completion (autocomplete) for arguments to simplify the user experience. i3lock-color currently does not. Tab autocomplete would make both testing and command-line usage easier.

## Task
Add tab completion for arguments. To illustrate this, if I type `i3lock --key<tab>` then it should show me
```
--keylayout mode
--keyhlcolor=rrggbbaa
```
if I continue with `l<tab>` (i.e. now I have typed i3lock --keyl<tab>`), I will get `keylayout` autocompleted for me. That is, I will now have "typed" `i3lock --keylayout`.

This enhancement is a nice and rewarding first issue, so if anyone wants to jump into i3lock-color/C/Linux development, this is a great starting point!

This seems like a nontrivial task:
- https://stackoverflow.com/questions/6107438/c-console-application-auto-complete-dynamic-arguments
- https://www.gnu.org/savannah-checkouts/gnu/bash/manual/bash.html#Programmable-Completion-Builtins

### PR DESCRIPTION
**Pull request by JezerM**
Closes #204

## Description
 This PR adds tab completion for bash and zsh

### Screenshots/screencaps

ZSH completion 1
ZSH completion 2
BASH completion

## Release notes

Notes: Adding tab completion for bash and zsh

### CODE REVIEW

File: install-i3lock-color.sh [hunk 1]
```diff
@@ -6,3 +6,11 @@ cd build
 sudo make install
 
 sudo install -Dm644 ../LICENSE "/usr/share/licenses/i3lock-color/LICENSE"
+
```
Comment (JezerM) [inline install-i3lock-color.sh:9]: I don't actually know how all the make process works here, so I made this workaround

File: install-i3lock-color.sh [hunk 2]
```diff
@@ -6,3 +6,11 @@ cd build
...
+if [[ -f /usr/bin/zsh ]]; then
+  sudo cp ../_i3lock /usr/share/zsh/vendor-completions
+fi
+
+if [[ -f /usr/bin/bash ]]; then
+  sudo cp ../i3lock-bash /usr/share/bash-completion/completions/i3lock
+fi
```
Comment (JezerM) [inline install-i3lock-color.sh]: I don't actually know how all the make process works here, so I made this workaround

**Comment by Raymo111** [comment]
Wow... this is massive. Thanks a bunch for contributing, I'll check this out sometime soon!

File: install-i3lock-color.sh (hunk 2 above)
Comment (Raymo111) [reply install-i3lock-color.sh]: Make uses m4 (GNU autotools and automake). Yes it's very annoying. I'm going to integrate this with m4 before I merge it. Thanks again for contributing :slightly_smiling_face:

Comment (JezerM) [reply install-i3lock-color.sh]: Well, I played around with autotools/automake and moved the completions to it :D

**Review by Raymo111 — APPROVED** [review]
LGTM

File: install-i3lock-color.sh (hunk 2 above)
Comment (Raymo111) [reply install-i3lock-color.sh]: Nice. Thanks for that :)

**Pull request merged**